File: 1-50/problem2/problem2.scala
val fibs: Stream[BigInt] =
    BigInt(0) #:: BigInt(1) #:: fibs.zip(fibs.tail).map { n => n._1 + n._2 }
val sum = fibs.takeWhile(_ < 4000000).filter(_ % 2 == 0).sum
println(sum)

File: 1-50/problem2/problem2.py
import itertools

def fib():
    a, b = 0, 1
    while True:
        yield a
        a, b = b, a + b

is_even = lambda x: x % 2 == 0
nums = itertools.takewhile(lambda x: x < 4000000, fib())
print(sum(filter(is_even, nums)))